### PR TITLE
Allow override of default styling

### PR DESCRIPTION
**Pull request by joel100**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -71,38 +71,52 @@ var FloatLabelTextField = React.createClass({
     };
   },
 
-  withBorder: function() {
-    if (!this.props.noBorder) {
-      return styles.withBorder;
-    };
+  componentWillReceiveProps: function(newProps) {
+    console.log('componentWillReceiveProps');
```
Comment (eyaleizenberg) [inline index.js]: remove

File: index.js [hunk 2]
```diff
@@ -71,38 +71,52 @@ var FloatLabelTextField = React.createClass({
...
+    });
+  },
+
+  getAdditionalStyle: function(name) {
+    if (this.props.style && this.props.style[name])
+    {
+      return this.props.style[name];
+    }
+
+    return {};
```
Comment (eyaleizenberg) [inline index.js:86]: you don't need to return anything if there is no style.
In fact you can probably do `return this.props.style && this.props.style[name]`

File: index.js [hunk 3]
```diff
@@ -123,28 +137,36 @@ var FloatLabelTextField = React.createClass({
...
 
   placeHolderValue: function() {
     if (this.state.text) {
-      return this.props.placeHolder;
+      return this.props.placeholder;
     }
   },
 
   setText: function(value) {
+    console.log(`SetText: ${value}`);
```
Comment (eyaleizenberg) [inline index.js]: remove

File: index.js [hunk 4]
```diff
@@ -123,28 +137,36 @@ var FloatLabelTextField = React.createClass({
...
-    } catch (_error) {}
+
+    this.textInput.refs.input.setNativeProps(value);
   },
 
-  withMargin: function() {
-    if (this.state.text) {
-      return styles.withMargin;
+  onChange: function(event) {
+    console.log(`onChange: ${event.nativeEvent.text}`);
```
Comment (eyaleizenberg) [inline index.js]: remove

File: index.js [hunk 5]
```diff
@@ -171,28 +189,26 @@ var styles = StyleSheet.create({
   fieldLabel: {
     height: 10,
     fontSize: 9,
-    color: '#B1B1B1'
+    color: '#B1B1B1',
+    paddingLeft: 15
```
Comment (eyaleizenberg) [inline index.js:193]: why are you adding padding here?

File: index.js [hunk 6]
```diff
@@ -171,28 +189,26 @@ var styles = StyleSheet.create({
...
     flex: 1,
     justifyContent: 'center',
-    position: 'relative'
-  },
-  withBorder: {
+    position: 'relative',
     borderBottomWidth: 1 / 2,
     borderColor: '#C8C7CC',
+    backgroundColor: 'white',
+    paddingLeft: 15
```
Comment (eyaleizenberg) [inline index.js:202]: again, why?

File: LICENSE [hunk 7]
```diff
@@ -1,6 +1,7 @@
 The MIT License (MIT)
 
-Copyright (c) 2015 Eyal Eizenberg
+Copyright (c) 2015 Eyal Eizenberg,
+Copyright (c) 2016 Joel McCormick
```
Comment (eyaleizenberg) [inline LICENSE:4]: ?

**Comment by eyaleizenberg** [comment]
1. See comments.
2. Can't merge because of conflicts.
3. Can you please provide an example of how you would use this?

File: index.js (hunk 5 above)
Comment (joel100) [inline index.js:193]: So that you don't need the View with style padding view.. More flexible

File: index.js (hunk 6 above)
Comment (joel100) [inline index.js:202]: Same reason

File: index.js (hunk 2 above)
Comment (joel100) [inline index.js:86]: I don't think returning a bool will help here .. What behavior is adding a null or undefined style going to cause now or in future in react?

File: LICENSE (hunk 7 above)
Comment (joel100) [inline LICENSE:4]: Your code was a nice example but wasn't really usable outside of the one use case. I've changed over 25% of the file to make it useful more broadly so I think this is reasonable.

**Comment by joel100** [comment]
Here is an example of how it is right now. I have kept the styling as it was internally which has worked for me. It might be worth creating a translation between a flat style style for the component and what is used internally.

```
<FloatingLabelTextInput placeholder="test" 
                                          style={{
                                            fieldContainer: { 
                                              backgroundColor: 'white',
                                              borderRadius: 3,
                                              borderColor: 'gray',
                                              borderWidth: 1,
                                              paddingLeft: 7 }, 
                                            fieldLabel: { paddingLeft: 7 }, 
                                            focussed: { color: cssV('dark-gray') },
                                            container: { height: 42 },
                                            valueText: { 
                                              fontSize: 17,
                                              fontFamily: 'Helvetica',
                                              color: 'gray' }}}/>
```

I've created another branch that would accept flat styling for a limited number of styles. Have a look and let me know which you prefer.
https://github.com/joel100/react-native-floating-label-text-input/tree/flat_styling

Example of flat styling for above branch:

```
<FloatingLabelTextInput placeholder="test" 
                                       style={{
                                          backgroundColor: 'white',
                                          borderRadius: 3,
                                          borderColor: 'gray',
                                          borderWidth: 1,
                                          paddingLeft: 7,
                                          focussedColor: cssV('dark-gray'),
                                          color: 'gray',
                                          height: 42, 
                                          fontSize: 16,
                                          fontFamily: 'Helvetica'}}/>
```

File: README.md [hunk 8]
```diff
@@ -34,6 +34,14 @@ var SomeComponent = React.createClass({
 ```
 
 ### Component props
+<<<<<<< HEAD
```
Comment (eyaleizenberg) [inline README.md]: you left conflict resolution

File: index.js (hunk 2 above)
Comment (eyaleizenberg) [inline index.js:86]: nulls are dropped in style.

**Comment by eyaleizenberg** [comment]
Can you create a PR for the flat style? I want to compare. Thanks.

**Pull request closed without merging**